runtime/builtin_types.go

```diff
@@ -559,6 +560,33 @@ func builtinRange(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException)
 	return ListType.Call(f, []*Object{r}, nil)
 }
 
+func builtinRawInput(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if len(args) > 1 {
+		msg := fmt.Sprintf("[raw_]input expcted at most 1 arguments, got %d", len(args))
+		return nil, f.RaiseType(TypeErrorType, msg)
+	}
+
+	stdout := os.Stdout
+	stdin := NewFileFromFD(os.Stdin.Fd())
```

Comment:
The trouble with creating a new File wrapping os.Stdin is that it will have a different buffered reader than the one held by sys.stdin. This is especially problematic because this File object is cleaned up after builtinRawInput() returns, meaning that whatever buffered data it held just disappears.

This goes back to my suggestion of having grumpy.Stdin/out/err variables that get used by this function and having sys.stdin/out/err be aliases for those variables:

```
# sys.py
from __go__.grumpy import Stdin as stdin, Stdout as stdout, Stderr as stderr
```

This way any buffering done for one call will be used by subsequent calls.